packages/ui-app/src/util/isValidBalance.ts

```diff
@@ -0,0 +1,237 @@
+
+    return {
+      isValid: false,
+      errorMessage: t(errorMessageKey, {
+        defaultValue: errorMessageUntranslated
+      }),
+      errorMessageUntranslated
+    };
+  }
+
+  // always a string from <input type='number'> but leave as failsafe
+  if (!(typeof input === 'string')) {
+    throw Error(t('balance.error.string.required', {
+      defaultValue: 'Balance input value must be valid type'
+    }));
+  }
+
+  // impossible since using <input type='number'> and prevents spaces but leave as failsafe
+  input = input.toLowerCase().split(' ').join('');
+
+  const matchE = input.match(/e/gi);
+  const matchPlus = input.match(/\+/gi);
+  const matchEPlus = input.match(/e\+/gi);
+
+  const matchEBeforeNumberOrPlus = input.match(/(e)(?=[0-9+])/ig);
+  const matchEPlusBeforeNumber = input.match(/(e\+)(?=[0-9])/ig);
+
+  if (matchE && matchE.length > 1 || matchPlus && matchPlus.length > 1) {
+    errorMessageKey = 'balance.error.notation.scientific.duplicate';
+    errorMessageUntranslated = 'Scientific notation may only contain one instance of \'e\' for scientific notation or exponential with \'e+\'';
+
+    return {
+      isValid: false,
+      errorMessage: t(errorMessageKey, {
+        defaultValue: errorMessageUntranslated
+      }),
+      errorMessageUntranslated
+    };
+  }
+
+  if (matchEPlus && matchEPlus.length === 1 && !matchEPlusBeforeNumber) {
+    errorMessageKey = 'balance.error.notation.exponential.exponent.missing';
+    errorMessageUntranslated = 'Exponential \'e+\' notation when used must be followed by a number';
+
+    return {
+      isValid: false,
+      errorMessage: t(errorMessageKey, {
+        defaultValue: errorMessageUntranslated
+      }),
+      errorMessageUntranslated
+    };
+  }
+
+  if (matchE && matchE.length === 1 && !matchEBeforeNumberOrPlus) {
+    errorMessageKey = 'balance.error.notation.scientific.exponent.missing';
+    errorMessageUntranslated = 'Scientific notation \'e\' when used must be followed by a number';
+
+    return {
+      isValid: false,
+      errorMessage: t(errorMessageKey, {
+        defaultValue: errorMessageUntranslated
+      }),
+      errorMessageUntranslated
+    };
+  }
+
+  // check the string only contains integers digits or scientific notation
+  if (!reValidInputChars.test(input)) {
+    errorMessageKey = 'balance.error.format';
+    errorMessageUntranslated = 'Balance to transfer in DOTs must be a number or expressed in scientific notation (i.e. 3.4e38) or exponential with \'e+\'';
+
+    return {
+      isValid: false,
+      errorMessage: t(errorMessageKey, {
+        defaultValue: errorMessageUntranslated
+      }),
+      errorMessageUntranslated
+    };
+  }
+
+  // check value is not decimal and greater than or equal to one, whether it be scientific notation, exponential (which allow decimal)
+  if (Number(parseFloat(input)) < 1 && Number(parseFloat(input)) !== 0) {
```

Comment:
We are using BN for a reason - JS cannot handle these. (Until BigInt is available)